proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,108 @@
+# Custom url resolution: Draft 1
+
+_[(Issue)](https://github.com/sass/sass/issues/2535)_
+
+## Table of Contents
+
+- [Background](#background)
+- [Summary](#summary)
+  - [Steps](#steps)
+  - [JavaScript API](#javascript-api)
+- [Semantics](#semantics)
+  - [Using Variables](#using-variables)
+- [Deprecation Process](#deprecation-process)
+
+## Background
+
+> This section is non-normative.
+
+Many css features require the use of a url reference to reference resources from outside the sass files, however these files also need to exist on the eventual output directory and server. To ensure the references are valid, the sass API should allow for the user to provide a way to remap and/or inline these resources.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a standardized way to remap the url references to the final location on the server or inline reference.
+
+This is accomplished by running url references through the url rewriting plugin if a url rewriting plugin has been defined.
+
+### Steps
+
+The steps of rewriting a url reference:
+
+- Url reference gets extracted from the sass file (For more information see [Semantics](#semantics))
+- These reference gets passed into the rewrite url plugin which returns the output url of this resource.
+- This returned value gets used to replace the original url reference.
+
+### JavaScript API
+
+At the core of remapping the url references is the JavaScript API which allows users to return a new url reference, based on filepath and original url reference.
+
+The first parameter of the urlRewrite function is an object with the following values:
+
+- `file`: The file path of the sass file that references the url.
+- `entry`: The entry file path that has been used to call the sass compiler.
```

Comment:
What's the use case for having access to the entrypoint?

Reply:
For outputting relative paths?

It's not really important though, as it can be worked around it by using a wrapper function or something that gets passed in additional options...

Just seemed cleaner to have it in the params, if you don't like it I can remove it

Reply:
You're right, will remove it for now we can append the outFile in the future if we want to. Third party tooling like bundlers have this information already anyway